### PR TITLE
Add common modules into PowerShell

### PR DESCRIPTION
**Pull request by maddieclayton**
## Description
https://github.com/Azure/azure-powershell/issues/7339

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
sign job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign-esrp/70/

**Comment by maddieclayton** [comment]
sign job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign-esrp/71/

File: src/Az.Test.props [hunk 1]
```diff
@@ -25,6 +25,11 @@
     <DefineConstants>TRACE;RELEASE;NETSTANDARD</DefineConstants>
   </PropertyGroup>
 
+  <ItemGroup>
+    <ProjectReference Condition="'$(IncludeCommonTestProject)' != 'false'" Include="$(MSBuildThisFileDirectory)\..\tools\ScenarioTest.ResourceManager\ScenarioTest.ResourceManager.csproj" />
```
Comment (MiYanni) [inline src/Az.Test.props]: Move the `Condition="'$(IncludeCommonTestProject)' != 'false'"` to the ItemGroup and remove it from each ProjectReference. Then, change the paths to `Include="$(RepoTools)ScenarioTest.ResourceManager\ScenarioTest.ResourceManager.csproj"` for each.

File: build.proj [hunk 2]
```diff
@@ -219,7 +219,7 @@
     <Exec Command="dotnet --version" />
     <Exec Command="dotnet new sln -n Azure.PowerShell --force" />
     <ItemGroup>
-      <ModuleCsprojFiles Include="$(LibraryRoot)src/**/*.csproj" Exclude="$(LibraryRoot)src/**/*.Test.csproj" />
+      <ModuleCsprojFiles Include="$(LibraryRoot)src/**/*.csproj" Exclude="$(LibraryRoot)src/**/*.Test.csproj;$(LibraryRoot)src/**/ScenarioTest.ResourceManager.csproj;$(LibraryRoot)src/**/TestFx.csproj" />
```
Comment (MiYanni) [inline build.proj]: Remove this change since these projects are moved to `tools`.

File: src/Accounts/Accounts/Az.Accounts.psd1 [hunk 3]
```diff
@@ -77,7 +75,9 @@ RequiredAssemblies = '.\Microsoft.Azure.PowerShell.Authentication.Abstractions.d
                '.\Microsoft.WindowsAzure.Storage.dll', 
                '.\Microsoft.WindowsAzure.Storage.DataMovement.dll', 
                '.\Microsoft.Azure.PowerShell.Clients.Aks.dll', 
-               '.\Microsoft.Azure.PowerShell.Strategies.dll'
+               '.\Microsoft.Azure.PowerShell.Strategies.dll',
+               '.\Microsoft.Azure.Commands.Common.Authentication.dll',
+               '.\Microsoft.Azure.Commands.Common.Authentication.ResourceManager.dll'
```
Comment (MiYanni) [inline src/Accounts/Accounts/Az.Accounts.psd1]: These should not be changed.

File: tools/ScenarioTest.ResourceManager/ScenarioTest.ResourceManager.csproj [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <RootNamespace>$(LegacyAssemblyPrefix)ScenarioTest</RootNamespace>
+    <IncludeCommonTestProject>false</IncludeCommonTestProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Moq" Version="4.7.145" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Accounts\Authentication\Authentication.csproj" />
```
Comment (MiYanni) [inline tools/ScenarioTest.ResourceManager/ScenarioTest.ResourceManager.csproj]: Please use `Include="$(RepoSrc)Accounts\Authentication\Authentication.csproj". Same for the other project reference.

File: tools/TestFx/TestFx.csproj [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <IncludeCommonTestProject>false</IncludeCommonTestProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Moq" Version="4.7.145" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\ScenarioTest.ResourceManager\ScenarioTest.ResourceManager.csproj" />
+    <ProjectReference Include="..\..\src\Accounts\Authentication\Authentication.csproj" />
```
Comment (MiYanni) [inline tools/TestFx/TestFx.csproj]: Please use `Include="$(RepoSrc)Accounts\Authentication\Authentication.csproj". Same for the other project reference.

**Pull request merged**